### PR TITLE
feat(run-tests): attempt to get version to make decisions on what tes…

### PR DESCRIPTION
**Pull request by Avantol13**
…ts to run

### CODE REVIEW

File: run-tests.sh [hunk 1]
```diff
@@ -31,6 +31,57 @@ dryrun() {
...
+  # Get last item of delimited string using string operators:
+  #    https://www.linuxjournal.com/article/8919
+  version=${response##*:}
+  echo $version
+}
+
+# Takes 3 arguments:
+#   $1 service name
+#   $2 test tag to avoid until version is greater than next arg
+#   $3 versio of service where tests apply >=
```
Comment (paulineribeyre) [inline run-tests.sh]:
```suggestion
#   $3 version of service where tests apply >=
```

File: run-tests.sh [hunk 2]
```diff
@@ -31,6 +31,57 @@ dryrun() {
...
+#
+# ex: runServiceTestsIfVersion "fence" "@multipartupload" "3.0.0"
```
Comment (paulineribeyre) [inline run-tests.sh]: not important, but the order of the arguments bothers me a little bit haha
`runServiceTestsIfVersion "@multipartupload" "fence" "3.0.0"` would look more natural to me

File: run-tests.sh [hunk 3]
```diff
@@ -31,6 +31,57 @@ dryrun() {
...
+runServiceTestsIfVersion() {
+  local currentVersion
+  currentVersion=$(getServiceVersion $1)
```
Comment (paulineribeyre) [inline run-tests.sh]: idk much about bash style conventions but should you make sure the arguments are provided?

File: run-tests.sh (hunk 2 above)
Comment (Avantol13) [reply run-tests.sh]: I did think it seems a little weird, I'll change it to your suggestion

File: run-tests.sh [hunk 4]
```diff
@@ -135,8 +176,23 @@ EOM
...
 
+# environments that use DCF features
+# we only run Google Data Access tests for cdis-manifest PRs to these
+envsRequireGoogle="dcp.bionimbus.org gen3.datastage.io nci-crdc-staging.datacommons.io nci-crdc.datacommons.io"
+
+#
+# DataClientCLI tests require a fix to avoid parallel test runs
+# contending over config files in the home directory
+#
+doNotRunRegex="@dataClientCLI"
```
Comment (paulineribeyre) [inline run-tests.sh]:
```suggestion
donot '@dataClientCLI'
```
`doNotRunRegex` should be initialized earlier

File: run-tests.sh [hunk 5]
```diff
@@ -56,8 +100,7 @@ donot() {
   doNotRunRegex="${doNotRunRegex}${or}$1"
```
Comment (paulineribeyre) [inline run-tests.sh]: mm i was wrong :p `doNotRunRegex` does not need to be initialized

**Comment by paulineribeyre** [comment]
a cool improvement would be to allow multiple service versions: for rbac you need latest fence, indexd and arborist

**Pull request merged**